Vendor tasecureapi

jcarrete5 opened the pull request:
PR Checklist:

- [x] Link to related issues: #24 
- [x] Add changelog entry linking to issue
- [x] Add tests (if needed)
- [x] If new feature: added in description / readme

jcarrete5 committed:
Avoid using unmaintained actions for installing the toolchain

jcarrete5 committed:
WIP: Use cc

jcarrete5 committed:
Replace CMake with `cc` and vendor YAJL

Closes [#24](https://github.com/rdkcentral/secapi-rust/issues/24)

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 21 files pending identification.
* Protex Server Path: /home/blackduck/github/secapi-rust/29/rdkcentral/secapi-rust

* Commit: cc0077ced3cd56278553eb37b36525ea8ce3ab44

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/6d328c8a65b7f52d84fafb4e41e2c64d)'

mhughesacn commented:
Hi Jason: Please will you follow the standard RDK way of incorporating opensourced software:
- Remove the COPYING file in this PR.  We do not like separate licenses in most cases.
- But append the contents of COPYING to the end of LICENSE at top level.  Please use a generic copyright instead of Lloyd H:
`Copyright <YEAR> <COPYRIGHT HOLDER>
`  so that the license is generally re-useable.
- And in NOTICE, please append:
```
yajl is:
Copyright (c) 2007-2014, Lloyd Hilaiel <me@lloyd.io> et al.
Licensed under the ISC License
```
I can then clear off the BD and Copyright votes.  I will ignore for now as the PR is draft.
Thank you

jcarrete5 committed:
Fix up copyright notice

mhughesacn commented:
Hi @jcarrete5 : Thank you for making these changes.  For clarity, please give the ISC license a header in LICENSE i.e. just add "ISC License" before the text of the license.

About your question, our focus is on what is hosted in this repo.  So NOTICE should credit only what is here.

jcarrete5 committed:
Add "ISC License" before license text

rdkcmf-jenkins commented:
b'## Copyright scan failure
 **Commit:** 4238585a8a137b484380834650b2c80e7632cfe1
**Report detail:**  https://gist.github.com/rdkcmf-jenkins/975ece47bbac5f4e06e4c1ddfcfd941f'

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 20 files pending identification.
* Protex Server Path: /home/blackduck/github/secapi-rust/29/rdkcentral/secapi-rust

* Commit: 4238585a8a137b484380834650b2c80e7632cfe1

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/ebf4a57fe0fab4e9ccf4d99f9060a644)'

jcarrete5 committed:
Merge branch 'main' into 24-vendor-tasecureapi

rdkcmf-jenkins commented:
b'## Copyright scan failure
 **Commit:** cafcdc9c9328e378062a4edc348fca0c9b7549de
**Report detail:**  https://gist.github.com/rdkcmf-jenkins/e7f4612b8a6441c221ce5b99bb8c0832'

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 20 files pending identification.
* Protex Server Path: /home/blackduck/github/secapi-rust/29/rdkcentral/secapi-rust

* Commit: cafcdc9c9328e378062a4edc348fca0c9b7549de

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/fe0dea735769f27d220cf4026b757177)'

jcarrete5 committed:
Update CI workflow

jcarrete5 committed:
Update tasecureapi

jcarrete5 committed:
Add ENABLE_SVP define

jcarrete5 committed:
Remove root_keystore.p12

This file is now embedded in the tasecureapi library and no longer needs
to be provided.

jcarrete5 committed:
Include Cargo.lock

jcarrete5 commented on LICENSE:
```diff
@@ -198,4 +198,20 @@
+
+ISC License
+
+Copyright (c) <YEAR>, <COPYRIGHT HOLDER>
```
It is supposed to contain placeholder text. The copyright info is called out elsewhere.

jcarrete5 committed:
Add rerun-if-changed directives

These directives will trigger build.rs to rebuild when the vendored
sources change.

jcarrete5 commented on secapi-sys/build.rs:
```diff
@@ -16,66 +16,291 @@
+    let include_dir = path!(yajl_build / "include");
+
+    std::fs::create_dir_all(path!(include_dir / "yajl")).unwrap();
+    for header in path!("yajl" / "api").read_dir().unwrap() {
```
addressed
